include/seqan3/argument_parser/validators.hpp

```diff
@@ -309,68 +328,260 @@ class file_ext_validator
      */
-    template <std::ranges::ForwardRange range_type>
-    //!\cond
-        requires std::ConvertibleTo<value_type_t<range_type>, std::filesystem::path const &>
-    //!\endcond
-    void operator()(range_type const & v) const
+    void validate_path_impl(std::filesystem::path const & path) const
     {
-         std::for_each(v.begin(), v.end(), [&] (auto cmp) { (*this)(cmp); });
+        if (path.has_extension() || !std::filesystem::exists(path))
+            throw parser_invalid_argument(detail::to_string("The directory ", path, " does not exist or is a file."));
     }
+};
+
+/*!\brief A validator that checks if a given path is a valid output directory.
+ * \ingroup argument_parser
+ * \implements seqan3::validator_concept
+ *
+ * \details
+ *
+ * The struct than acts as a functor that throws a seqan3::parser_invalid_argument exception whenever a given directory
+ * (std::filesystem::path) already exist or the specified path is a filename.
+ *
+ * \snippet test/snippet/argument_parser/validators_output_directory.cpp usage
+ *
+ * \note The validator works on every type that can be implicitly converted to std::filesystem::path.
+ */
+class output_directory_validator : public path_validator_base<output_directory_validator>
+{
+private:
+
+    //!\brief The type of the base class.
+    using base_t = path_validator_base<output_directory_validator>;
+
+    //!\brief Befriends the base class.
+    friend base_t;
+
+    //!\brief Inherit the base constructor.
+    using base_t::base_t;
+
+public:
+    //!\brief Type of values that are tested by validator.
+    using base_t::value_type;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    output_directory_validator() = default;                                               //!< Defaulted.
+    output_directory_validator(output_directory_validator const &) = default;             //!< Defaulted.
+    output_directory_validator(output_directory_validator &&) = default;                  //!< Defaulted.
+    output_directory_validator & operator=(output_directory_validator const &) = default; //!< Defaulted.
+    output_directory_validator & operator=(output_directory_validator &&) = default;      //!< Defaulted.
+    ~output_directory_validator() = default;                                              //!< Defaulted.
+    //!\}
 
     //!\brief Returns a message that can be appended to the (positional) options help page info.
     std::string get_help_page_message() const
     {
-        return detail::to_string("The file or directory is checked for existence.");
+        return detail::to_string("The output directory must not exist and end with a valid directory separator.");
+    }
+
+private:
+
+    /*!\brief Checks if directory is valid.
+     * \param path The directory name to check.
+     * \throws parser_invalid_argument
+     */
+    void validate_path_impl(std::filesystem::path const & path) const
+    {
+        if (path.has_extension())
```

Comment:
Shouldn't an output directory also exist?

Reply:
Well, now the output directory might or might not exists. It will be checked in both cases if it is writable.